src/wp-includes/block-bindings/sources/post-meta.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Add the post_meta source to the Block Bindings API.
+ *
+ * @since 6.5.0
+ * @package WordPress
+ */
+function post_meta_source_callback( $source_attrs ) {
+	// Use the postId attribute if available
+	if ( isset( $source_attrs['postId'] ) ) {
+		$post_id = $source_attrs['postId'];
+	} else {
+		// $block_instance->context['postId'] is not available in the Image block.
+		$post_id = get_the_ID();
```

Comment:
I think there might be a bug here. 

Instead of completely omitting the usage of the $block_instance->context['postId'] here, we should check if the value isset and if so use it over the ID we get from `get_the_ID`.

Something like this:

```php
$post_id = isset( $block_instance->context['postId'] ) ? $block_instance->context['postId'] : get_the_ID();
```

Reply:
I also wonder if we should always use the context here. as if I'm not wrong the default context should be set as `get_the_ID` if there's no parent query block.

Reply:
From what I tested, `$block_instance->context` is an empty array for the heading, button, and image block, so that's why we couldn't get the `postId` from there. Is that not expected?

We could definitely use the conditional suggested. However, I'd like to understand if just using `$block_instance->context` should work.

Reply:
Maybe we have an issue somewhere but in the client the block context has some "default context values" like the post ID if I'm not wrong, and in the server you're saying that these are not present. IMO, the block context should be the same for the client and the server. 

I think it's fine if we defer the potential fix to a dedicated issue but we might want to track it somewhere.